content/blog/elevate-your-v-project-with-unit-tests.md

```diff
@@ -0,0 +1,749 @@
+---
+title: "Elevate Your V Project With Unit Tests"
+subtitle: V makes it easy to add tests to your project. Here's how to get started.
+summary: The next blog post after The Complete Beginner's Guide to CLI Apps in V. We discuss why unit testing is important, shed some light on V's testing culture, and add unit tests to our toy project geo.
+page: Blog
+authorname: Subhomoy Haldar
+authorlink: subhomoy-haldar
+categories: ["tutorials"]
+date: 2022-01-24T14:59:44+05:30
+image: images/elevate-your-v-projects-with-unit-tests/hero.webp
+---
+
+## Previous Article
+
+This article picks up after the previous one:
+[The Complete Beginner's Guide to CLI Apps in V](/the-complete-beginners-guide-to-cli-apps-in-v/).
+If you haven't already, we encourage you to read that article first.
+
+Afterwards, you can obtain the full source code on
+[GitHub](https://github.com/hungrybluedev/geo) and continue with this
+article.
+
+## Why Even Bother?
+
+Let's start by addressing the question: _Why even bother with unit tests?_
+Here are a few reasons why unit tests make sense:
+
+1. **Unit tests are easy to write.** It's a preventitive measure that prevents
+   a lot of bugs from reaching the final product.
+2. In order to write effective unit tests, we need to organise our codebase
+   into **separate, independent, isolated modules**. This way, we can ensure
+   that each of the smaller pieces work correctly on their own, before we try
+   to check the entire system as a whole. It saves a lot of time and effort.
+3. Splitting the system into manageable pieces also makes it easier to focus
+   on **improving the quality of the code independently**, without having to
+   rely on the other parts of the system. A team can assign different people
+   to different parts of the project and they can all work independently,
+   while being confident that things will be fine as long as their code passes
+   all the unit tests.
+4. Running a unit test (ideally) does not take a long time. This **speeds up**
+   **the iteration process** and helps developers ship their software faster.
```

Comment:
Point 1 is not very logical too - unit tests may be easy or hard to write, but that is not related directly to the next sentence "It's a preventitive measure that prevents a lot of bugs from reaching the final product."
Something like this may be better, not sure:
```
1. **Unit tests are easy to write, and can prevent a lot of bugs from reaching the final product.**
```

Points 2-3 are not relevant imho, and are better left out - they are more about the organization of a project (small mostly independent modules), not about testing specifically. One can benefit from testing, even without subscribing to these organizational principles. 

Point 4 may also be reformulated like this:
```
4. Running a unit test usually does not take long, unlike other more extensive kinds of tests.
```
because whether it will speed up, or slow down iterations is also debatable (depends very much on the kind of the work, the experience of the developers, how good the documentation of the used libraries is etc).